pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -1961,6 +1965,22 @@ func (r *resourceManager) SetSecrets(s Secrets) { r.secrets = s }
 // GetValues returns the current configuration values of the deployer.
 func (r *resourceManager) GetValues() Values { return r.values }
 
+// GetManagedResourceName returns the name for the ManagedResource containing resources deployed to the target cluster.
+func (r *resourceManager) GetManagedResourceName() string {
+	if r.values.NamePrefix == "" {
+		return shootManagedResourceName
+	}
+	return r.values.NamePrefix + v1beta1constants.DeploymentNameGardenerResourceManager
+}
+
+// GetTargetAccessSecretName is the name of the target access secret for the gardener-resource-manager.
+func (r *resourceManager) GetTargetAccessSecretName() string {
+	if r.values.NamePrefix == "" {
+		return secretNameShootAccess
+	}
+	return fmt.Sprintf("%saccess-%s", r.values.NamePrefix, v1beta1constants.DeploymentNameGardenerResourceManager)
+}
```

Comment:
So, you are suggesting that we keep the `shoot-access-...` naming of the secrets for the operator in the `garden` namespace? That is certainly easier in terms of code change. As long as we do not get confused by the names of the resulting artefacts that should be fine.

Reply:
Yes, that's the suggestion. Obviously, I agree that having something more generic than seed/shoot (like runtime/target) is "more correct"/better, and if there were easy ways to achieve this, I'd be the first to go there, however, as said before, I don't think those exist. Doing this "renaming" consistently is a lot of effort, and the current changes just make it more inconsistent. At least for the time being, I propose to not change this and finish the `gardener-operator` epic for now. Maybe we can work on consistent renaming in the aftermath.

Reply:
If consistent naming is not the first priority, we don't need to introduce `GetManagedResourceName()` method to the resource manager, but keep `shoot-core-gardener-resource-manager` as static name of the managed resource instead.
This would simplify the deployment of resource manager. WDYT?

Reply:
as discussed, we keep the static `shoot-...` names for the sake of simplicity